de.dlr.sc.virsat.model.extension.requirements.ui/src/de/dlr/sc/virsat/model/extension/requirements/ui/wizard/CsvFileReqTypeSelectionPage.java

```diff
@@ -187,59 +183,57 @@ public boolean isSelectionValid() {
-				& !dataNumberStartField.getText().equals("")) {
-
-				final String destination = getDestination();
-	
-				if (isSelectionValid()) {
-					CategoryAssignment selection = (CategoryAssignment) getSelection();
-					Repository repository = VirSatResourceSet.getVirSatResourceSet(selection).getRepository();
-					ActiveConceptHelper activeConceptHelper = new ActiveConceptHelper(repository);
-					Concept activeReqConcept = activeConceptHelper.getConcept(de.dlr.sc.virsat.model.extension.requirements.Activator.getPluginId());
+		if (isDestinationSelected 
+				&& isCurrentPage() 
+				&& !headerNumberField.getText().equals("")
+				&& seperatorField.getText().length() == 1   //Only allow exactly one char
+				&& !dataNumberStartField.getText().equals("")
+				&& isSelectionValid()) {
+
+			CategoryAssignment selection = (CategoryAssignment) getSelection();
+			Repository repository = VirSatResourceSet.getVirSatResourceSet(selection).getRepository();
+			ActiveConceptHelper activeConceptHelper = new ActiveConceptHelper(repository);
+			Concept activeReqConcept = activeConceptHelper
+					.getConcept(de.dlr.sc.virsat.model.extension.requirements.Activator.getPluginId());
+			
+			// Configure CSV reader
+			final String destination = getDestination();
+			CsvFileReader reader = getWizard().getReader();
+			reader.setSeparator(getSeparator());
+			reader.setHeaderLine(getHeaderLineNumber());
+			
+			//Do the parsing
+			if (!reader.parseFile(destination)) {
+				Status status = new Status(Status.ERROR, Activator.getPluginId(),
+						"CSVImportWizard: Failed to perform parsing of CSV file!");
+				StatusManager.getManager().handle(status, StatusManager.LOG | StatusManager.SHOW);
+				return false;
+			} 
 			
-					try {
-						wizard.getReader().setSeparator(getSeparator());
-						wizard.getReader().setHeaderLine(getHeaderLineNumber());
-						csvHeader = wizard.getReader().readCsvHeadline(destination);
-					} catch (IOException e) {
-						Status status = new Status(Status.ERROR, Activator.getPluginId(),
-								"CatiaImportWizard: Failed to perform import! Selected file not valid!", e);
-						StatusManager.getManager().handle(status, StatusManager.LOG | StatusManager.SHOW);
-					}
-					
-					if (selection.getType().getFullQualifiedName().equals(RequirementType.FULL_QUALIFIED_CATEGORY_NAME)) {
-						typeReviewPage.setInput(csvHeader, new RequirementType(selection));
-					}
-					if (selection.getType().getFullQualifiedName().equals(RequirementsConfiguration.FULL_QUALIFIED_CATEGORY_NAME)) {
-						reqType = wizard.getImporter().prepareRequirementType(activeReqConcept, csvHeader);
-						typeReviewPage.setInput(csvHeader, reqType);
-					}
-					
-					return true;
-				}
-				
+			csvHeader = reader.readCsvHeadline();
+			if (selection.getType().getFullQualifiedName().equals(RequirementType.FULL_QUALIFIED_CATEGORY_NAME)) {
+				typeReviewPage.setInput(csvHeader, new RequirementType(selection));

```

Comment:
Because we want differnt behavior depending on what kind of element was selected. If a configration container was selcted then we want our importer to create the type for us... Then selection is not event of type RequirementsType...

Reply:
Sure, I understand this, I mean when the selection is RequirementType, we are creating two different beans on the same underlying CA (which is a selection).
Shall we maybe move this line `typeReviewPage.setInput(csvHeader, reqType);` after the ifs?

Reply:
Ah yes, didn't see that